Sources/TuistDependencies/DependenciesController.swift

```diff
@@ -150,3 +157,13 @@ public final class DependenciesController: DependenciesControlling {
         }
     }
 }
+
+extension DependenciesGraph {
+    fileprivate func merging(with other: Self) throws -> Self {
+        let mergedThirdPartyDependencies = try thirdPartyDependencies.merging(other.thirdPartyDependencies) { old, _ in
+            let name = self.thirdPartyDependencies.first { $0.value == old }!.key
```

Comment:
Since every `ThirdPartyDependency` will have a name, maybe it's something we ought to save in that enum? My thinking is that we might have to do this dance where we have the enum but don't have the name at multiple places and adding that to the enum would make it much more streamlined and less error-prone

Reply:
Makes sense to me, so you would add the name to `ThirdPartyDependency`, and the property in the `DependencyGraph` would become a Set, right? 👍

Reply:
I'd be wary about making it a `Set` because we'd rely on `Hashable` conformance for `ThirdPartyDependency` consisting of name only. However, that is not completely right IMHO and I'd probably reserve the hashable implementation for caching - for which we will need to add e.g. the path of the dependency, too.

Reply:
Then it probably make sense to keep the dictionary and have the name both in the key and in the value 👌